### PR TITLE
Do not offer 'convert' namespace when the ns has sibling types

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/74214

### CODE REVIEW

File: src/Features/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs [hunk 1]
```diff
@@ -19,27 +19,20 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertNamespace;
...
 
     [Fact]
     public async Task TestNoConvertToFileScopedInCSharp9()
     {
-        var code = @"
-namespace $$N
-{
-}
-";
+        var code = """
```
Comment (CyrusNajmabadi) [inline src/Features/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs:31]: just converting to raw srtings.

File: src/Features/DiagnosticsTestUtilities/CodeActions/CSharpCodeFixVerifier`2+Test.cs [hunk 2]
```diff
@@ -61,6 +62,9 @@ public Test()
             /// <inheritdoc cref="SharedVerifierState.Options"/>
             internal OptionsCollection Options => _sharedState.Options;
 
+            [StringSyntax("C#-Test")] public new string TestCode { set => base.TestCode = value; }
+            [StringSyntax("C#-Test")] public new string FixedCode { set => base.FixedCode = value; }
```
Comment (CyrusNajmabadi) [inline src/Features/DiagnosticsTestUtilities/CodeActions/CSharpCodeFixVerifier`2+Test.cs:66]: allows syntax highlighting of test code.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CompilerExtensions.projitems [hunk 3]
```diff
@@ -244,6 +244,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)EmbeddedLanguages\Common\EmbeddedSyntaxToken.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)EmbeddedLanguages\Common\EmbeddedSyntaxTree.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)EmbeddedLanguages\Common\EmbeddedSyntaxTrivia.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)EmbeddedLanguages\PredefinedEmbeddedLanguageClassifierNames.cs" />
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CompilerExtensions.projitems:247]: moved to shared file so we can reference the "CSharpTest" constant in the test files.

File: src/Features/CSharp/Portable/ConvertNamespace/ConvertNamespaceCodeRefactoringProvider.cs [hunk 4]
```diff
@@ -24,14 +24,10 @@ namespace Microsoft.CodeAnalysis.CSharp.ConvertNamespace;
 using static ConvertNamespaceTransform;
 
 [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertNamespace), Shared]
-internal class ConvertNamespaceCodeRefactoringProvider : SyntaxEditorBasedCodeRefactoringProvider
+[method: ImportingConstructor]
+[method: SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+internal class ConvertNamespaceCodeRefactoringProvider() : SyntaxEditorBasedCodeRefactoringProvider
```
Comment (ToddGrun) [inline src/Features/CSharp/Portable/ConvertNamespace/ConvertNamespaceCodeRefactoringProvider.cs:29]: nit: seal?

**Review by ToddGrun — APPROVED** [review]

**Review by akhera99 — APPROVED** [review]

**Pull request merged**